crates/rpc/rpc-eth-api/src/helpers/transaction.rs

```diff
@@ -643,3 +646,58 @@ pub trait LoadTransaction: SpawnBlocking {
         }
     }
 }
+
+/// Find a transaction by sender and nonce
+pub async fn get_transaction_by_sender_and_nonce<Eth>(
+    eth: &Eth,
+    sender: Address,
+    nonce: u64,
+    include_pending: bool,
+) -> Result<Option<RpcTransaction<Eth::NetworkTypes>>, Eth::Error>
+where
+    Eth: EthBlocks + EthState,
+{
```

Comment:
this should be a trait method of `EthTransactions`
```rust
/// Find a transaction by sender and nonce
fn get_transaction_by_sender_and_nonce<Eth>(
    eth: &Eth,
    sender: Address,
    nonce: u64,
    include_pending: bool,
) -> impl Future<Output = Result<Option<RpcTransaction<Eth::NetworkTypes>>, Eth::Error>> + Send
where
    Self: LoadBlock + LoadState,
```

see other comments for instructions how to relax trait bound from `EthBlocks + EthState` to ` LoadBlock + LoadState`

Reply:
@emhane thanks for the detailed information.